resharper/resharper-yaml/src/Daemon/Stages/YamlDaemonStageBase.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using JetBrains.Application.Settings;
+using JetBrains.ReSharper.Feature.Services.Daemon;
+using JetBrains.ReSharper.Plugins.Yaml.Psi;
+using JetBrains.ReSharper.Plugins.Yaml.Psi.Tree;
+using JetBrains.ReSharper.Psi;
+using JetBrains.ReSharper.Psi.Files;
+using JetBrains.Util;
+
+namespace JetBrains.ReSharper.Plugins.Yaml.Daemon.Stages
+{
+  public abstract class YamlDaemonStageBase : IDaemonStage
+  {
+    public IEnumerable<IDaemonStageProcess> CreateProcess(IDaemonProcess process, IContextBoundSettingsStore settings,
+      DaemonProcessKind processKind)
+    {
+      if (!IsSupported(process.SourceFile))
+        return EmptyList<IDaemonStageProcess>.InstanceList;
+
+      process.SourceFile.GetPsiServices().Files.AssertAllDocumentAreCommitted();
+
+      return process.SourceFile.GetPsiFiles<YamlLanguage>()
+        .SelectNotNull(file => CreateProcess(process, settings, processKind, (IYamlFile) file));
+    }
+
+
+    protected abstract IDaemonStageProcess CreateProcess(IDaemonProcess process, IContextBoundSettingsStore settings,
+      DaemonProcessKind processKind, IYamlFile file);
+
+    public ErrorStripeRequest NeedsErrorStripe(IPsiSourceFile sourceFile, IContextBoundSettingsStore settingsStore)
+    {
```

Comment:
What's this for and why is this method required?

Reply:
The perils of an initial commit style PR :) This method used to be on `IDaemonStage`, but has been removed. I've already removed it in following commits.